jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1002,6 +1002,15 @@ typedef enum
   ECMA_GC_COLOR__COUNT /**< number of colors */
 } ecma_gc_color_t;
 
+#ifndef CONFIG_ECMA_PROPERTY_HASHMAP_DISABLE
+typedef enum
+{
+  ECMA_PROP_HASHMAP_ALLOC_ON,       /**< property hashmap allocation is enabled */
+  ECMA_PROP_HASHMAP_ALLOC_TEMP_OFF, /**< property hashmap allocation is temprary disabled */
+  ECMA_PROP_HASHMAP_ALLOC_OFF       /**< property hashmap allocation is disabled */
+} ecma_property_hashmap_allocation_state_t;
+#endif /* !CONFIG_ECMA_PROPERTY_HASHMAP_DISABLE */
```

Comment:
I would like a counter for this. The number should be between 0-4.

0 - allocation allowed
>0 - allocation disallowed

JMEM_FREE_UNUSED_MEMORY_SEVERITY_LOW -> decrease the number
JMEM_FREE_UNUSED_MEMORY_SEVERITY_HIGH -> increase the number

if JMEM_FREE_UNUSED_MEMORY_SEVERITY_HIGH happens twice in a row increase the counter to 4.